front-end/src/components/gallery/Gallery.js

```diff
@@ -1,33 +1,89 @@
 import axios from "axios";
 import { useEffect, useState } from "react";
 import styled from 'styled-components';
+import Filter from "../filter/Filter";
 import Product from '../product/Product';
+import DisplayFilters from '../filter/DisplayFilters';
 
 
 const Gallery = () => {
+    const [filters, setFilters] = useState('');
     const [products, setProducts] = useState('');
     const [currPage, setCurrPage] = useState(1);
+    const [categoryOption, setCategoryOption] = useState({});
+    const [sizeOption, setSizeOption] = useState([]);
+    const [colorOption, setColorOption] = useState([]);
 
     useEffect(() => {
-        axios.get(`//localhost:9999/products?page=${currPage}&limit=12`, {
-            params: {
-                test: 'test'
-            }
-        })
+        const categoryString = new URLSearchParams(categoryOption);
+        const categoryUrl = `&${categoryString.toString()}`;
+        let sizeString = `&size=${sizeOption.toString()}`;
+        let colorString = `&color=${colorOption.toString()}`;
+
+        axios.get(`//localhost:9999/products?page=${currPage}&limit=12${categoryUrl}${sizeString}${colorString}`)
             .then(res => {
                 setProducts(res.data)
             })
             .catch(err => console.log(err));
-    }, [currPage]);
+    }, [currPage, categoryOption, sizeOption, colorOption]);
+
+    useEffect(() => {
+        axios.get(`//localhost:9999/productsFilters`)
+            .then(res => {
+                setFilters(res.data)
+            })
+            .catch(err => console.log(err));
+    }, [categoryOption]);
+
+    const removeCategory = () => {
+        setCategoryOption({})
+    }
+
+    const removeFilter = (filterType, value) => {
+        if (filterType === 'size') {
+            const newArray = sizeOption.filter(i => i !== value)
```

Comment:
Use descriptive variable names instead of `i` so the code is more self-documenting for anyone else in the future
```suggestion
            const newArray = sizeOption.filter(size => size !== value)
```